python/ray/serve/tests/test_deploy.py

```diff
@@ -1193,6 +1195,140 @@ async def __call__(self):
     assert requests.get(url).text == "2"
 
 
+class TestDeployGroup:
+    @serve.deployment
+    def f(request=None):
+        return "f reached"
+
+    @serve.deployment
+    def g(request=None):
+        return "g reached"
+
+    @serve.deployment
+    class C:
+        async def __call__(self):
+            return "C reached"
+
+    @serve.deployment
+    class D:
+        async def __call__(self):
+            return "D reached"
+
+    @serve.deployment
+    class E:
+        async def __init__(self):
+            self.F_handle = serve.get_deployment("F").get_handle()
+
+        async def __call__(self):
+            return await self.F_handle.request_output.remote()
+
+        async def request_output(self):
+            return await self.F_handle.get_output.remote()
+
+        async def get_output(self):
+            return "E reached"
+
+    @serve.deployment
+    class F:
+        async def __init__(self):
+            self.E_handle = serve.get_deployment("E").get_handle()
+
+        async def __call__(self):
+            return await self.E_handle.request_output.remote()
+
+        async def request_output(self):
+            return await self.E_handle.get_output.remote()
+
+        async def get_output(self):
+            return "F reached"
+
+    @serve.deployment
+    class MutHandles:
+        async def __init__(self, handle_name):
+            self.handle = serve.get_deployment(handle_name).get_handle()
+
+        async def __call__(self, echo: str):
+            return await self.handle.request_echo.remote(echo)
+
+        async def request_echo(self, echo: str):
+            return echo
+
+    @serve.deployment(num_replicas=2, max_concurrent_queries=5)
+    class DecoratedClass1:
+        async def __call__(self):
+            return "DecoratedClass1 reached"
+
+    @serve.deployment(num_replicas=4, max_concurrent_queries=2)
+    class DecoratedClass2:
+        async def __call__(self):
+            return "DecoratedClass2 reached"
+
+    def deploy_and_check_responses(self, deployments, responses):
```

Comment:
these tests are quite hard to understand without context due to all the redirection -- could you add header comments to each with a high-level description?